### ISSUE TITLE
Git isn't working for me

### ISSUE THREAD

**Issue by cabalamat** [mblayman/markwiki#37]
I set up a config file with

    GIT_ENABLED = True

Then I ran `markwiki`. I got this error message:

```
*** Please tell me who you are.

Run

  git config --global user.email "you@example.com"
  git config --global user.name "Your Name"
```

I have already set up Git on my system so was a bit surprised. I set it up again, and ran markwiki again, same error message.

Over and above the error I'm having, can I suggest that, in line with the principle "MarkWiki aims to be dead simple", that if a Git name/email aren't set, defaults be used.

**Comment by mblayman** [comment]
I'm not an active user of MarkWiki so I can't speak much to this feature.

@deadc0de6, is the Git feature still working for you? As the implementer of the Git stuff, I'm wondering if you have any idea what is going on.

**Comment by cabalamat** [comment]
> that if a Git name/email aren't set, defaults be used

I've had a look at the code and `git_integration.py` has:

```
    def _set_config(self):
        '''Set git configs.'''
        self.git.config('--local', 'core.autocrlf', 'input')
        self.git.config('--local', 'user.name', 'markwiki')
        self.git.config('--local', 'user.email', '')
```

So it looks like it already does this and I'm surprised it's not working for me. I'll have another look this evening.

**Comment by deadc0de6** [comment]
The git config should be setup with defaults (https://github.com/mblayman/markwiki/blob/master/markwiki/git/git_integration.py#L27)

I tried fresh and it worked for me.
Here's the walkthrough I used if it can be of any help:

```
$ git clone https://github.com/mblayman/markwiki.git /tmp/markwiki
$ cat > /tmp/markwiki/production.config 
AUTHENTICATION = None
ADMINISTRATOR = None
ADMIN_PASSWORD = None
ALLOW_REGISTRATION = True
SECRET_KEY = 'It\'s a secret to everybody.'
STORAGE_TYPE = 'file'
MARKWIKI_HOME = '/tmp/the-wiki'
DEBUG = False
STATIC_PATH = None
TEMPLATE_PATH = None
GIT_ENABLED = True
$ cd /tmp/markwiki
$ virtualenv env
(env) $ source env/bin/activate
(env) $ python setup.py install
(env) $ MARKWIKI_SETTINGS="/tmp/markwiki/production.config" markwiki
...
(env) $ cat /tmp/the-wiki/wiki/.git/config 
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	autocrlf = input
[user]
	name = markwiki
	email = 
```

git seems to be setup and working. I have also tested the creation and modification of pages and it works, it is commit'ed and the history can be viewed.

Could you detailed how you set it up and your config file so I can test it on my side.

**Comment by cabalamat** [comment]
How I set it up: I used `git clone` to put it in `~/sproj/markwiki/`. From that directory I ran:
```
phil:~/sproj/markwiki$ echo $MARKWIKI_SETTINGS
/home/phil/markwikirc
phil:~/sproj/markwiki$ virtualenv venv
New python executable in venv/bin/python
Installing setuptools, pip...done.
phil:~/sproj/markwiki$ . venv/bin/activate
(venv)phil:~/sproj/markwiki$ python setup.py develop
(venv)phil:~/sproj/markwiki$ markwiki
```

It is at this point I get the error message above.

My `markwikirc` is:
```
DEBUG = True
STATIC_PATH = None
TEMPLATE_PATH = None
GIT_ENABLED = True
```

NB a `~/.markwiki/wiki/.git/` directory is being created and the git repository works from the command line:
```

phil:~/.markwiki/wiki$ git status
On branch master

Initial commit

Changes to be committed:
  (use "git rm --cached <file>..." to unstage)

        new file:   Home.md
        new file:   MarkWiki/Deployment.md

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        MarkWiki/FAQ.md
        MarkWiki/Features.md
        MarkWiki/Help.md
        MarkWiki/Introduction.md
        MarkWiki/Settings.md
```

FWIW I'm using git 1.9.1

**Comment by cabalamat** [comment]
I have also tried manually entering my details into `~/.markwiki/wiki/.git/config`, but that doesn't work because whenever I do that and subsequently start `markwiki`, the `config` file is over-written.

**Comment by deadc0de6** [comment]
Indeed, the empty email field was not accepted by git. In my case it was using git global config for the empty field. Starting fresh gave me the same error as you.
The above PR should fix it.

**Comment by cabalamat** [comment]
Thanks

**Comment by mblayman** [comment]
Fixed by #38 

Thanks for reporting *and* fixing, guys.
